apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,370 @@
+	TC_ASSERT_GEQ("task_manager_register_task", tm_sample_handle2, 0);
+
+	int ret;
+
+	ret = task_manager_start(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_start", ret, OK, unregister_tasks_1());
+
+	ret = task_manager_start(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_start", ret, OK, unregister_tasks_1());
+
+	unregister_task1 = false;
+	sleep(1);
+
+	if (unregister_task1) {
+		ret = task_manager_unregister(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+		TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+	}
+
+	ret = task_manager_unregister(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+void cleanup_task5(void)
+{
+	unregister_task5 = true;
+}
+
+void itc_task_manager_pause_task2(void)
+{
+	int ret;
+	int tm_sample_handle4;
+	tm_appinfo_list_t *sample_list_info;
+
+	sample_list_info = ((tm_appinfo_list_t *)task_manager_getinfo_with_name("task5", TM_RESPONSE_WAIT_INF));
+	TC_ASSERT_NEQ("task_manager_getinfo_with_name", sample_list_info, NULL);
+
+	tm_sample_handle4 = sample_list_info->task.handle;
+	task_manager_clean_infolist(&sample_list_info);
+	TC_ASSERT_NEQ("task_manager_clean_infolist", (tm_appinfo_list_t *)&sample_list_info, NULL);
+
+	ret = task_manager_pause(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_pause", ret, OK, cleanup_task5());
+
+	ret = task_manager_resume(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_resume", ret, OK, cleanup_task5());
+
+	ret = task_manager_restart(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_restart", ret, OK, cleanup_task5());
+
+	ret = task_manager_stop(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_stop", ret, OK, cleanup_task5());
+
+	ret = task_manager_unregister(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_unregister", ret, OK, cleanup_task5());
+
+}
+
+int itc_task_manager_pause_task2_main(int argc, char *argv[])
+{
+	itc_task_manager_pause_task2();
+	return 0;
+}
+
+void itc_task_manager_task4(void)
+{
+	int tm_sample_handle4;
+
+	tm_sample_handle4 = task_manager_register_task("task5", 100, 1024, not_builtin_task_1, NULL, TM_APP_PERMISSION_ALL, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register_task", tm_sample_handle4, 0);
+
+	unregister_task5 = false;
+	int ret;
+
+	ret = task_manager_start(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_register", ret, OK);
+
+	while (1) {
+		usleep(1);
+		if (unregister_task5) {
+			ret = task_manager_unregister(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
+			TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+		}
+	}
+}
+
+int itc_task_manager_task4_main(int argc, char *argv[])
+{
+	itc_task_manager_task4();
```

Comment:
is there any special reason that calling itc_task_manager_task4() separately, not in this function?

Reply:
No reason